lib/internal/modules/esm/translators.js

```diff
@@ -511,3 +484,28 @@ translators.set('wasm', async function(url, source) {
     }
   }).module;
 });
+
+// Strategy for loading a MTS module
+translators.set('module-typescript', async function(url, source) {
+  assertBufferSource(source, false, 'load');
+  const code = parseTScode(source);
+  debug(`Translating TypeScript ${url}`);
+  return FunctionPrototypeCall(translators.get('module'), this, url, code, false);
+});
+
+// Strategy for loading a TypeScript module
+translators.set('typescript', async function(url, source) {
+  assertBufferSource(source, false, 'load');
+  const code = parseTScode(source);
+  debug(`Translating TypeScript ${url}`);
+  // Try to guess if the TypeScript is a CTS or MTS
+  // Depends on --experimental-detect-module to be unflagged
```

Comment:
We shouldn’t depend on detection as the only or even the primary way of determining the module system for TypeScript files. We should determine whether to treat a `.ts` file as CommonJS or as ESM the same way we determine how to treat a `.js` file: the `package.json` `type` field, falling back to detection.

Reply:
since for now it only supports esm, I removed the detection